### PR TITLE
add es _stats API

### LINKED ISSUE #3526: Support `indices-stats` elasticsearch endpoint
[Elasticsearch doc ref](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/indices-stats.html)

Many fields returned by the endpoint do not make sense for Quickwit. For this issue, we will only focus on the following fields:

- `stats.primaries.docs.count`: number of docs in published splits
- `stats.primaries.size_in_bytes`: size in bytes of published  splits
- `stats.total.segments.count`: number of published splits

### PR DESCRIPTION
**Pull request by PSeitz**
add /{index}/_stats API
add /_stats API

closes #3526

### CODE REVIEW

File: quickwit/quickwit-serve/src/elasticsearch_api/filter.rs [hunk 1]
```diff
@@ -171,6 +171,20 @@ pub(crate) fn elastic_index_count_filter(
...
+pub(crate) fn elastic_index_stats_filter(
+) -> impl Filter<Extract = (Vec<String>,), Error = Rejection> + Clone {
+    warp::path!("_elastic" / String / "_stats")
+        .and_then(extract_index_id_patterns)
+        .and(warp::get().or(warp::post()).unify())
+}
+
+#[utoipa::path(get, tag = "Search", path = "/_stats")]
+pub(crate) fn elastic_stats_filter() -> impl Filter<Extract = (), Error = Rejection> + Clone {
+    warp::path!("_elastic" / "_stats").and(warp::get().or(warp::post()).unify())
```
Comment (fmassot) [inline quickwit/quickwit-serve/src/elasticsearch_api/filter.rs:185]: do we need to support POST request?

Comment (PSeitz) [reply quickwit/quickwit-serve/src/elasticsearch_api/filter.rs]: no, I removed it

**Review by fmassot — APPROVED** [review]

File: quickwit/rest-api-tests/scenarii/es_compatibility/0020-stats.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+method: [GET]
+engines:
+  - quickwit
+  - elasticsearch
+endpoint: "gharchive/_stats"
+expected:
```
Comment (fmassot) [inline quickwit/rest-api-tests/scenarii/es_compatibility/0020-stats.yaml:6]: Can we add `size_in_bytes` in the expected values?

Comment (PSeitz) [reply quickwit/rest-api-tests/scenarii/es_compatibility/0020-stats.yaml:6]: I'm not sure why, but the value is not stable

Comment (fmassot) [reply quickwit/rest-api-tests/scenarii/es_compatibility/0020-stats.yaml:6]: This is really strange, I would open an issue to track this unexpected behavior :/

**Pull request merged**